.github/workflows/cpp.yml

```diff
@@ -114,7 +114,11 @@ jobs:
           ulimit -c unlimited
           archery docker run ${{ matrix.image }}
       - name: Docker Push
-        if: success() && github.event_name == 'push' && github.repository == 'apache/arrow'
+        if: >-
+          success() &&
```

Comment:
We might be able to reduce code duplication a bit by refactoring this into an action (similar to #35896 )

Reply:
It's a good idea!

Reply:
I've created an issue to track it: https://github.com/apache/arrow/issues/36235